test/db_test.py

```diff
@@ -0,0 +1,358 @@
+
+		@it('None returned for non-existing user')
+		def noUserFound():
+			user = db.getUser('badname')
+			assert_that(user, none())
+
+	@describe('Add User')
+	def addUser():
+
+		@afterEach
+		def _afterEach():
+			cleanup()
+
+		@it('Fields preserved')
+		def fieldsPreserved():
+			db.addUser(test_user)
+			row = db.DB_CONN.execute('''
+				SELECT name, pass_hash, pass_salt, email
+				FROM Users WHERE name = ?
+			''', [test_name]).fetchone()
+
+			assert_that(row[0], equal_to(test_name))
+			assert_that(row[1], equal_to(test_hash))
+			assert_that(row[2], equal_to(test_salt))
+			assert_that(row[3], equal_to(test_email))
+
+		@it('Dates auto-populated')
+		def datesPopulated():
+			db.addUser(test_user)
+			row = db.DB_CONN.execute('''
+				SELECT created_at, updated_at, accessed_at
+				FROM Users WHERE name = ?
+			''', [test_name]).fetchone()
+
+			assertDateNearNow(row[0])
+			assertDateNearNow(row[1])
+			#assertDateNearNow(row[2])
+
+		@it('Cannot have multiple users with the same name')
+		def duplicateName():
+			db.addUser(test_user)
+			assert_that(
+				calling(db.addUser).with_args(test_user),
+				raises(IntegrityError, 'UNIQUE constraint failed: Users.name')
+			)
+
+		@it('Password hash and salt required')
+		def hashAndSaltRequired():
+			assert_that(
+				calling(db.addUser).with_args({
+					'name': test_name,
+					'pass_hash': test_hash,
+					'pass_salt': None
+				}),
+				raises(IntegrityError, 'NOT NULL constraint failed: Users.pass_salt')
+			)
+			assert_that(
+				calling(db.addUser).with_args({
+					'name': test_name,
+					'pass_hash': None,
+					'pass_salt': test_salt
+				}),
+				raises(IntegrityError, 'NOT NULL constraint failed: Users.pass_hash')
+			)
+
+	@describe('Update User')
+	def updateUser():
+
+		added_user = None
+
+		@beforeEach
+		def _beforeEach():
+			nonlocal added_user
+			row = addTestUser(whole_row=True)
+			added_user = {
+				'id': row[0],
+				'name': row[1],
+				'pass_hash': row[2],
+				'pass_salt': row[3],
+				'email': row[4],
+				'created_at': row[5],
+				'updated_at': row[6],
+				'accessed_at': row[7]
+			}
+
+		@afterEach
+		def _afterEach():
+			cleanup()
+
+		@it('Update user email preserves other fields')
+		def updatedUser():
+			nonlocal added_user
+			update_email = 'new@email.com'
+			added_user['email'] = update_email
+			res = db.updateUser(added_user)
+			updated_user = db.getUser(test_name)
+
+			assert_that(updated_user.get('name'), equal_to(test_name))
+			assert_that(updated_user.get('pass_hash'), equal_to(test_hash))
+			assert_that(updated_user.get('pass_salt'), equal_to(test_salt))
+			assert_that(updated_user.get('email'), equal_to(update_email))
+
+		@it('Date modified changed on update')
+		def modifiedUpdated():
+			sleep(1) # Delay to ensure modified time is different
+			db.updateUser(test_user)
+			update_user = db.getUser(test_name)
+			assert_that(
+				update_user.get('updated_at'),
+				not_(equal_to(test_user.get('updated_at')))
```

Comment:
When is this value (test_user : 'updated_at') set? It seems that updated_at is not set until after the call on line 199. If so, this test cannot fail.